"ansible-playbook setup.yml" fails on fresh vagrant install

After running through the steps in the setup guide, I reached the point where I needed to run "ansible-playbook setup.yml".

Each NPM install takes around 10 minutes, which seems high for installing things like grunt-cli, gulp, and bower. But the main issue I run into is that ansible fails on the "npm install" task with the following:

	npm ERR! Error: EPERM, chown '/home/vagrant/data/Attache/node_modules/grunt-contrib-clean/package.json'
	npm ERR!  { [Error: EPERM, chown '/home/vagrant/data/Attache/node_modules/grunt-contrib-clean/package.json']
	npm ERR!   errno: 50,
	npm ERR!   code: 'EPERM',
	npm ERR!   path: '/home/vagrant/data/Attache/node_modules/grunt-contrib-clean/package.json',
	npm ERR!   fstream_finish_call: 'chown',
	npm ERR!   fstream_type: 'File',
	npm ERR!   fstream_path: '/home/vagrant/data/Attache/node_modules/grunt-contrib-clean/package.json',
	npm ERR!   fstream_class: 'FileWriter',
	npm ERR!   fstream_stack: 
	npm ERR!    [ '/usr/lib/nodejs/fstream/lib/writer.js:305:19',
	npm ERR!      '/usr/lib/nodejs/graceful-fs/polyfills.js:133:7',
	npm ERR!      'Object.oncomplete (fs.js:107:15)' ] }
	npm ERR! 
	npm ERR! Please try running this command again as root/Administrator.

This message appears several times for various grunt packages before ansible completely fails and terminates.

At this point, I run ansible again with:
    ansible-playbook setup.yml -vvv --limit @/Users/brian/setup.retry

This time it fails with a different error:

	TASK: [grunt build] *********************************************************** 
	<192.168.33.10> ESTABLISH CONNECTION FOR USER: vagrant
	<192.168.33.10> REMOTE_MODULE command grunt build --force chdir=data/Attache
	<192.168.33.10> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/Users/brian/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o IdentityFile=".vagrant/machines/default/virtualbox/private_key" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=10 192.168.33.10 /bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682 && chmod a+rx $HOME/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682 && echo $HOME/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682'
	<192.168.33.10> PUT /var/folders/2_/m0nq9ctd7jdc2nzzdgk17n780000gn/T/tmp60as4j TO /home/vagrant/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682/command
	<192.168.33.10> EXEC ssh -C -tt -v -o ControlMaster=auto -o ControlPersist=60s -o ControlPath="/Users/brian/.ansible/cp/ansible-ssh-%h-%p-%r" -o StrictHostKeyChecking=no -o IdentityFile=".vagrant/machines/default/virtualbox/private_key" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=vagrant -o ConnectTimeout=10 192.168.33.10 /bin/sh -c 'LANG=en_US.UTF-8 PATH=/home/vagrant/.gem/ruby/2.2.0/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin LC_CTYPE=en_US.UTF-8 /usr/bin/python /home/vagrant/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682/command; rm -rf /home/vagrant/.ansible/tmp/ansible-tmp-1430246717.2-254125110973682/ >/dev/null 2>&1'
	failed: [192.168.33.10] => {"changed": true, "cmd": ["grunt", "build", "--force"], "delta": "0:00:00.104257", "end": "2015-04-28 18:45:17.389959", "rc": 99, "start": "2015-04-28 18:45:17.285702", "warnings": []}
	stdout: grunt-cli: The grunt command line interface. (v0.1.13)
	
	Fatal error: Unable to find local grunt.
	
	If you're seeing this message, either a Gruntfile wasn't found or grunt
	hasn't been installed locally to your project. For more information about
	installing and configuring grunt, please see the Getting Started guide:
	
	http://gruntjs.com/getting-started
	
	FATAL: all hosts have already failed -- aborting

setup.yml has not been modified.

--- a/Vagrantfile
+++ b/Vagrantfile
@@ -8,5 +8,7 @@ Vagrant.configure(2) do |config|
   config.vm.synced_folder "../data", "/home/vagrant/data", type: "nfs"
   config.vm.provider :virtualbox do |vb|
     vb.customize ["modifyvm", :id, "--memory", 1024]
+    vb.customize ["modifyvm", :id, "--natdnshostresolver1", "on"]
+    vb.customize ["modifyvm", :id, "--natdnsproxy1", "on"]
   end
 end

--- a/setup.yml
+++ b/setup.yml
@@ -217,9 +217,13 @@
       command: forever list
       register: forever_list
       changed_when: false
+      sudo: no
     - name: start API ver2.0 server
-      command: forever start /home/vagrant/data/Attache-node-api/server.js
-      when: "forever_list.stdout.find('/home/vagrant/data/Attache-node-api/server.js') == -1"
+      # the -w flag reloads the server when app files have changed
+      # the --uid flag lets us assign a unique ID to the daemon so we can find it later in "forever list"
+      command: forever -w --uid "attacheNodeServerID" start server.js chdir=data/Attache-node-api
+      when: "forever_list.stdout.find('attacheNodeServerID') == -1"
+      sudo: no
 
     - name: create mysql table
       mysql_db: name=attache state=import target=/home/vagrant/data/Attache/sql/create.sql

--- a/setup.yml
+++ b/setup.yml
@@ -61,7 +61,7 @@
     - name: Install git
       apt: name=git
     - name: copy private key
-      copy: src={{private_key_path}} dest=/home/vagrant/.ssh/id_rsa owner=vagrant group=vagrant mode=0600    
+      copy: src={{private_key_path}} dest=/home/vagrant/.ssh/id_rsa owner=vagrant group=vagrant mode=0600
 
     #############################################
     # Setup node.js(grunt, gulp, bower, forever)
@@ -203,6 +203,7 @@
       command: composer install chdir=data/Attache
     - name: npm install
       npm: path=/home/vagrant/data/Attache
+      sudo: no
     - name: grunt build
       command: grunt build --force chdir=data/Attache
       sudo: no
@@ -211,6 +212,7 @@
 
     - name: npm install (API ver2.0)
       npm: path=/home/vagrant/data/Attache-node-api
+      sudo: no
     - name: check list of Node.js apps running
       command: forever list
       register: forever_list

